src/network/main.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (C) 2021 YADRO.
+ */
+
+#include "com/yadro/NetworkAdapter/Manager/server.hpp"
+#include "dbus.hpp"
+#include "adapter.hpp"
+
+#include <boost/algorithm/string.hpp>
+#include <phosphor-logging/log.hpp>
+#include <sdbusplus/server.hpp>
+
+#include <fstream>
+#include <streambuf>
+#include <string>
+
+#include <nlohmann/json.hpp>
+
+using namespace phosphor::logging;
+using namespace nlohmann;
+
+static constexpr const char* adaptersDataFile = "/var/lib/inventory/mac.json";
+
+using NetworkAdapterManagerServer = sdbusplus::server::object_t<
+    sdbusplus::com::yadro::NetworkAdapter::server::Manager>;
+
+class Manager : NetworkAdapterManagerServer
+{
+  public:
+    Manager(sdbusplus::bus::bus& bus) :
+        NetworkAdapterManagerServer(bus, dbus::netadpmgr::path), bus(bus)
+    {}
+    void rescan();
+
+  private:
+    sdbusplus::bus::bus& bus;
+    std::vector<std::shared_ptr<NetworkAdapter>> adapters;
+};
+
+/**
+ * @brief Find network adapters information and create corresponding dbus objects
+ *
+ */
+void Manager::rescan()
+{
+    std::ifstream dataFile(adaptersDataFile);
+    std::stringstream buffer;
+    if (!dataFile.is_open())
+    {
+        log<level::ERR>("failed to open file",
+                        entry("VALUE=%s", adaptersDataFile));
+        return;
+    }
+
+    buffer << dataFile.rdbuf();
```

Comment:
`buffer` looks excess. `nlohmann/json` can read the data right from `fstream` object.
https://github.com/nlohmann/json#tofrom-streams-eg-files-string-streams

Reply:
fixed